src/org/github/jamm/strategies/PreJava15SpecStrategy.java

```diff
@@ -0,0 +1,219 @@
+        super(memoryLayout, contendedClass, mayBeContendedValueMH);
+    }
+
+    /**
+     * Align the size of the fields.
+     * <p>Prior to JDK 15 the field blocks for each super class were aligned based using the object reference size.
+     * This method provides a hook for that logic.</p>
+     *
+     * @param sizeOfDeclaredFields the size of the fields for a class of the hierarchy
+     * @return the size of the class fields aligned.
+     */
+    protected long alignFieldBlock(long sizeOfDeclaredFields) {
+        return roundTo(sizeOfDeclaredFields, memoryLayout.getReferenceSize());
+    }
+
+    @Override
+    protected final int arrayBaseOffset(Class<?> type) {
+        return memoryLayout.getArrayHeaderSize();
+    }
+
+    @Override
+    public final long measureInstance(Class<?> type) {
+
+        long size = sizeOfFields(type, false);
+        return roundTo(size, memoryLayout.getObjectAlignment());
+    }
+
+    private long sizeOfFields(Class<?> type, boolean useFieldBlockAlignment) {
+
+        if (type == Object.class)
+            return memoryLayout.getObjectHeaderSize();
+
+        long size = sizeOfFields(type.getSuperclass(), true);
+
+        long sizeOfDeclaredField = 0;
+        long sizeTakenBy8BytesFields = 0;
+        ContentionGroupsCounter contentionGroupCounter = null;
+        for (Field f : type.getDeclaredFields()) {
+            if (!Modifier.isStatic(f.getModifiers())) {
+                int fieldSize = measureField(f.getType());
+
+                sizeTakenBy8BytesFields += fieldSize & 8; // count only the 8 bytes fields
+                sizeOfDeclaredField += fieldSize;
+                // If some fields are annotated with @Contended we need to count the contention groups to know how much padding needs to be added
+                if (CONTENDED_ENABLED)
+                    contentionGroupCounter = countContentionGroup(contentionGroupCounter, f);
+            }
+        }
+
+        // Ensure that we take into account the superclass gaps
+        if (hasSuperClassGap(size, sizeOfDeclaredField, sizeTakenBy8BytesFields)) {
+            size = roundTo(size, 8);
+        }
+
+        // Ensure that we take into account the hierarchy gaps
+        size += useFieldBlockAlignment ? alignFieldBlock(sizeOfDeclaredField) : sizeOfDeclaredField;
+
+        /* From Contended Javadoc:
+         * The class level {@code @Contended} annotation is not inherited and has
+         * no effect on the fields declared in any sub-classes. The effects of all
+         * {@code @Contended} annotations, however, remain in force for all
+         * subclass instances, providing isolation of all the defined contention
+         * groups. Contention group tags are not inherited, and the same tag used
+         * in a superclass and subclass, represent distinct contention groups.
+         */
+        if (CONTENDED_ENABLED && isClassAnnotatedWithContended(type))
+            size += (memoryLayout.getContendedPaddingWidth() << 1);
```

Comment:
Yes. The cast is not needed because we are adding to a long so the JVM will automatically perform it.

Reply:
I think in general the casting would be needed because `memoryLayout.getContendedPaddingWidth()` returns `int` and when we shift to the left it can become too big but in this case you are right the upper limit is 8192 so the max can be 16384 so it should be fine.